src/Examine.Lucene/Search/FacetDoubleRangeQueryField.cs

```diff
@@ -0,0 +1,27 @@
+using Examine.Search;
+
+namespace Examine.Lucene.Search
+{
+    public class FacetDoubleRangeQueryField : IFacetDoubleRangeQueryField
+    {
+        private readonly LuceneSearchQuery _search;
+        private readonly FacetDoubleField _field;
+
+        public FacetDoubleRangeQueryField(LuceneSearchQuery search, FacetDoubleField field)
```

Comment:
Shouldn't the interface be injected here instead of the concrete type `IFacetDoubleField` ? But then also wondering, is there a benefit of having these fields as interfaces? I understand having the underlying IFacetField since they all implement that, but looking through the code, all the type checking for IFacetField is done on the concrete types, not their interfaces. The PR could be simplified/reduced by removing the field interfaces apart from IFacetField?

Reply:
Then, the property setters for MaxCount and FacetField can be made internal on the concrete types

Reply:
Okay I moved `FacetQueryField` to the Core project and made `MaxCount` `internal set;`